extensions/src/poseHand/index.ts

```diff
@@ -0,0 +1,324 @@
+   */
+  // reset() {
+  // }
+
+  /**
+   * Checks if the hand pose estimate is ready to be used
+   * @returns {boolean} true if connected, false if not connected
+   */
+  isConnected() {
+    return !!this.handPoseState && this.handPoseState.length > 0;
+  }
+
+  /**
+   * Runs for the entire time the extension is running. Gets information about the video frame.
+   * Estimates where the hand is on the video frame. Creates a delay to prevent this function from constantly running,
+   * so as to prevent the entire program from slowing down.
+   */
+  async _loop() {
+    while (true) {
+      const frame = this.runtime.ioDevices.video.getFrame({
+        format: Video.FORMAT_IMAGE_DATA,
+        dimensions: PoseHand.DIMENSIONS
+      });
+
+      const time = +new Date();
+      if (frame) {
+        this.handPoseState = await this.estimateHandPoseOnImage(frame);
+        /*
+        if (this.isConnected()) {
+            this.runtime.emit(this.runtime.constructor.PERIPHERAL_CONNECTED);
+        } else {
+            this.runtime.emit(this.runtime.constructor.PERIPHERAL_DISCONNECTED);
+        }
+        */
+      }
+      const estimateThrottleTimeout = (+new Date() - time) / 4;
+      await new Promise(r => setTimeout(r, estimateThrottleTimeout));
+    }
+  }
+
+  /**
+   * Estimates where the hand is on the video frame.
+   * @param imageElement
+   * @returns {Promise<AnnotatedPrediction[]>}
+   */
+  async estimateHandPoseOnImage(imageElement) {
+    const handModel = await this.getLoadedHandModel();
+    return await handModel.estimateHands(imageElement, {
+      flipHorizontal: false
+    });
+  }
+
+  /**
+   * Gets the hand model from handpose
+   * @returns
+   */
+  async getLoadedHandModel() {
+    if (!this._handModel) {
+      this._handModel = await handpose.load();
+    }
+    return this._handModel;
+  }
+
+  /**
+   * Turns the video camera off/on/on and flipped. This is called in the operation of videoToggleBlock
+   * @param state 
+   */
+  videoToggle(state: number) {
+    if (state === VideoState.OFF) {
+      this.runtime.ioDevices.video.disableVideo();
+    }
+    else {
+      this.runtime.ioDevices.video.enableVideo();
+      // Mirror if state is ON. Do not mirror if state is ON_FLIPPED.
+      this.runtime.ioDevices.video.mirror = (state === VideoState.ON);
+    }
+  }
+
+  /**
+   * Sets the video's transparency. This is called in the operation of setVideoTransparencyBlock
+   * @param transparency 
+   */
+  setVideoTransparency(transparency: number) {
+    const trans = Math.max(Math.min(transparency, 100), 0);
+    this.runtime.ioDevices.video.setPreviewGhost(trans);
+  }
+
+  /**
+   * Sets up the default settings for the extension. Gives information related to each of the extension's blocks
+   * @returns The extension's blocks
+   */
+  defineBlocks(): PoseHand["BlockDefinitions"] {
+
+    /**
+     * Sets up the extension's default video settings
+     */
+    if (this.firstInstall) {
```

Comment:
Could you investigate if `defineBlocks` is ever called more than once? You can try adding the extension more than once, saving & loading a project with the extension when it's already added, etc. (let me know if you need any help). 

Because my suspicion is it never is, and thus I don't know why they implemented this logic. But maybe the original author knows something I don't!

Reply:
Great idea to checkout a new branch for that work, so you can freely add `console.log` and don't need to remember to clean it up.